src/utils/generateEventLink.ts

```diff
@@ -0,0 +1,102 @@
+import {
+    CalendarEvent as CalendarLinkEvent,
+    google,
+    outlook,
+    ics,
+} from 'calendar-link';
+import { parse } from 'date-fns';
+
+export const enum CalendarType {
+    Google,
+    Outlook,
+    ICS,
+}
+
+export interface CalendarEventDate {
+    day: number;
+    month: number;
+    year: number;
+    optional?: {
+        hours: number;
+        minutes: number;
+    };
+}
+
+export interface CalendarEvent {
+    title: string;
+    allDay: boolean;
+    location: string;
+    start: CalendarEventDate;
+    end?: CalendarEventDate;
+}
+
+/**
+ * Generates calendar link for an event by a calendar type
+ * @param event Event
+ * @param calendarType CalendarType
+ */
+export function generateEventLink(
+    event: CalendarEvent,
+    calendarType: CalendarType
+): string {
+    const calendarEvent = generateCalendarEvent(event);
+    switch (calendarType) {
+        case CalendarType.Google:
+            return google(calendarEvent);
+        case CalendarType.Outlook:
+            return outlook(calendarEvent);
+        default:
+            return decodeURIComponent(
+                ics(calendarEvent).split('charset=utf8,')[1]
+            );
+    }
+}
+
+/**
+ * in case of changing
```

Comment:
but improve what?

Reply:
The fact, that we need to change implementation and this constant - it breaks single responsibility rules and we need to avoid such boundaries as much as possible (change in place requires changes in other places).

In general it seems odd at first glance. I'd need to look more into packages APIs to understand what is going on there and what are possible solutions.

Reply:
Agree about that, but in such case, we will need to work with Regexp, which does not worth it from my perspective.
I can leave the comment for the future